File: 287-find-the-duplicate-number/287-find-the-duplicate-number.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        arr = [0] * (10**5 + 1)
        for num in nums:
            arr[num] += 1
        for a in range(len(arr)):
            if arr[a] > 1:
                return a

File: 287-find-the-duplicate-number/287-find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        const int n = 1e5 + 1;
        int vis[n];
        memset(vis, 0, sizeof(vis));
        for (auto x: nums) {
            if (vis[x])
                return x;
            vis[x]++;
        }
        return -1;
    }
};